hyperspy/io_plugins/emd.py

```diff
@@ -0,0 +1,392 @@
+            If True (default) loads data to memory. If False, enables loading only if requested.
+
+        Returns
+        -------
+        emd: :class:`~.EMD`
+            A :class:`~.EMD` object containing the loaded signals.
+
+        """
+        cls._log.debug('Calling load_from_ems')
+        # Read in file:
+        emd_file = h5py.File(filename, 'r')
+        # Creat empty EMD instance:
+        emd = EMD()
+        # Extract user:
+        user_group = emd_file.get('user')
+        if user_group is not None:
+            for key, value in user_group.attrs.iteritems():
+                emd.user[key] = value
+        # Extract microscope:
+        microscope_group = emd_file.get('microscope')
+        if microscope_group is not None:
+            for key, value in microscope_group.attrs.iteritems():
+                emd.microscope[key] = value
+        # Extract sample:
+        sample_group = emd_file.get('sample')
+        if sample_group is not None:
+            for key, value in sample_group.attrs.iteritems():
+                emd.sample[key] = value
+        # Extract comments:
+        comments_group = emd_file.get('comments')
+        if comments_group is not None:
+            for key, value in comments_group.attrs.iteritems():
+                emd.comments[key] = value
+        # Extract signals:
+        node_list = emd_file.keys()
+        for key in ['user', 'microscope', 'sample', 'comments']:  # Nodes which are not the data!
+            if key in node_list:
+                node_list.pop(node_list.index(key))  # Pop all unwanted nodes!
+        # One node should remain, the data node (named 'data', 'signals', 'experiments', ...)!
+        assert len(node_list) == 1, 'Dataset location is ambiguous!'
+        data_group = emd_file.get(node_list[0])
+        if data_group is not None:
+            for name, group in data_group.iteritems():
+                if isinstance(group, h5py.Group):
+                    if group.attrs.get('emd_group_type') == 1:
+                        emd._read_signal_from_group(name, group, load_to_memory)
+        # Close file and return EMD object:
+        if load_to_memory:
+            emd_file.close()
+        return emd
+
+    def save_to_emd(self, filename='datacollection.emd'):
+        """Save :class:`~.EMD` data in a file with emd(hdf5)-format.
+
+        Parameters
+        ----------
+        filename : string, optional
+            The name of the emd-file in which to store the signals.
+            The default is 'datacollection.emd'.
+
+        Returns
+        -------
+        None
+
+        """
+        self._log.debug('Calling save_to_emd')
+        # Open file:
+        emd_file = h5py.File(filename, 'w')
+        # Write version:
+        ver_maj, ver_min = EMD_VERSION.split('.')
+        emd_file.attrs['version_major'] = ver_maj
+        emd_file.attrs['version_minor'] = ver_min
+        # Write user:
+        user_group = emd_file.create_group('user')

```

Comment:
again, if overwriting only parts of the file, `require_group` is probably better.